src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -123,7 +123,7 @@ private StaticLangXCombinedSurfaceView generate(SurfaceTransformerContext contex
     view.grpcClientConstructorName(namer.getGrpcClientConstructorName(service));
 
     String outputPath = pathMapper.getOutputPath(service, apiConfig);
-    String fileName = GoSurfaceNamer.getReducedServiceName(service) + "_client.go";
+    String fileName = namer.getReducedServiceName(service).toLowerCamel() + "_client.go";
```

Comment:
Move this logic to GoSurfaceNamer.getServiceFileName(...), call SurfaceNamer.getServiceFileName from here.